### ISSUE TITLE
createUser requires "password", but real firebase does not

### ISSUE THREAD

**Issue by morgler** [soumak77/firebase-mock#108]
The mock does not seem to reflect the latest firebase interface. In my case, the `admin.auth().createUser()` method should work with any parameter. Especially the case `createUser({email: "some@email.com"})` should work (see documentation: https://firebase.google.com/docs/auth/admin/manage-users).

But `firebasemock.MockFirebase()` throws an error stating `Firebase.createUser failed: First argument must contain the key "password" with type "string"`. This is not in line with current firebase documentation.

If I run the `createUser({email: "some@email.com"})` on my production system, it throw NO error an produces a valid new user account. I would expect firebase-mock to reflect that behavior.

Please let me know, if I'm doing something wrong. Here is my test:

```
import * as admin from 'firebase-admin';

let mockfirestore = null

jest.mock('../src/firebase', () => {
  const firebasemock = require('firebase-mock');

  mockfirestore = new firebasemock.MockFirestore();
  mockfirestore.autoFlush();
  const mockauth = new firebasemock.MockFirebase();
  mockauth.autoFlush();
  const mocksdk = new firebasemock.MockFirebaseSdk(
    path => {
        return path ? mockdatabase.child(path) : mockdatabase;
    },
    () => {
        return mockauth;
    }
  );
  const firebase = mocksdk.initializeApp(); // can take a path arg to database url
  // optional - expose the mock
  global.firebase = firebase;

  return {
    db: mockfirestore,
    auth: mockauth
  }
})

...
// And in the test I just call my function named `func`

      await func(data, context)

```

**Comment by SidneyNemzer** [comment]
`firebase.auth()` doesn't have a `createUser` function, does it? https://firebase.google.com/docs/reference/js/firebase.auth.Auth

`admin.auth()` *does* have a `createUser()`, and indeed, the password is optional. But as far as I know, the client SDK cannot create a user without a password.

**Comment by soumak77** [comment]
Because the API is being used for both client and server testing, the createUser method was needed for both and as you called put, the required parameters differ between the client and server.

To get around this in my tests I just had my test code add a password for the server call even though it wasn't necessary. If you have a better suggestion for how to handle this I'll gladly take a PR.

**Comment by soumak77** [comment]
I can't remember which version it was, but in the past the client SDK had a `createUser` method.  It appears those have since been replaced with [createUserWithEmailAndPassword](https://firebase.google.com/docs/reference/js/firebase.auth.Auth#createUserWithEmailAndPassword) and [createUserAndRetrieveDataWithEmailAndPassword](https://firebase.google.com/docs/reference/js/firebase.auth.Auth#createUserAndRetrieveDataWithEmailAndPassword).

The admin SDK still has a [createUser](https://firebase.google.com/docs/reference/admin/node/admin.auth.Auth#createUser) method.

Therefore, we can remove the requirement for `createUser` to accept a password and introduce the two other client methods that require a password.

**Comment by soumak77** [comment]
fix available in v2.2.7

**Comment by morgler** [comment]
Wow, that was quick. Thanks :).
